### DISCUSSION TITLE
How to link against pinocchio when using pip install?

### DISCUSSION THREAD

**Discussion by lwygzh** [stack-of-tasks/pinocchio#2375] [Q&A] [answered]
Hi,

I am trying to create a Python package which contains C++ source code and its Python bindings. I am hoping to install this package using pip and put it on pypi. The C++ source code depends on `pinocchio` and when building the Python bindings on my computer with a system installed `pinocchio` package everything works nicely.

However, if I want to distribute my package via pypi, I don't think I can expect that people have `pinocchio` installed on their machine. I noticed that for packages that are pip installable and depends on `pinocchio`, such as `crocoddyl`, when running pip install they also download the `pinocchio` wheels from pypi. I was wondering if someone could help me understand how that `pinocchio` wheel is being used and how can we link against `pinocchio` when its pip installed, in other words, since its not installed in a default position what should I change to enable cmake to find `pinocchio`.

Any help is greatly appreciated!

**Comment by jcarpent** [comment, marked as answer]
Your best solution in this case is to use conda, not pip.

**Reply by lwygzh** [reply]
@jcarpent Thanks for the response!

I was trying to use `conda`, but I want to deploy my package on `linux-aarch64`. Some dependencies like `crocoddyl` does not support the architecture via conda but does support it via `pip`. So I planned to start with `pip`.

Thus, I would greatly appreciate some guidance on how to link against `pinocchio` when distributing the package using `pip.`

**Reply by cmastalli** [reply]
You could help us to support linux-aarch64 in conda. This should not be that difficult. More details at https://github.com/conda-forge/crocoddyl-feedstock/blob/main/conda-forge.yml

**Reply by lwygzh** [reply]
@cmastalli I can do that! I'll try to understand the process and do a PR. 

Just to check, assuming the build commands directly work, is it just adding `linux_aarch64: linux_64` to the `build_platform` as shown in the `pinocchio-feedstock`: https://github.com/conda-forge/pinocchio-feedstock/blob/main/conda-forge.yml?

**Reply by jcarpent** [reply]
Ongoing: https://github.com/conda-forge/crocoddyl-feedstock/pull/47

**Reply by cmastalli** [reply]
Thanks so much for your support, @jcarpent!

**Reply by jcarpent** [reply]
Solved and merged: https://github.com/conda-forge/crocoddyl-feedstock/pull/47

**Reply by lwygzh** [reply]
@jcarpent Thanks! I was looking into this, but did not fully understand how packaging in conda works when the code is Python bindings of C++ source code. I was wondering if you know any resources on this?

**Reply by jcarpent** [reply]
Gooogle it ;)
